source/agora/consensus/Fee.d

```diff
@@ -245,6 +245,10 @@ public class FeeManager
         if (tx_fee < required_fee)
             return "Transaction: There is not enough data fee";
 
+        tx_fee.div(tx.sizeInBytes());
+        if (tx_fee < params.MinFee)
+            return "Transaction: Fee is less than adjusted minimum fee";
```

Comment:
Two things:
- `div` returns an `Amount`. If the return is non-zero, it means that the condition should be `<=`.
- Wouldn't a better approach be:
```D
Amount minimumFee = params.MinFee;
if (!minimumFee.mul(tx.sizeInBytes())
    return "Fee: Transaction size overflows fee cap";
if (tx_fee < minimumFee)
    return "...";
```

Reply:
ugh, cannot count how many times I failed with this

Reply:
I think the original code is correct, but I might have misunderstood your comment.

minfee = 1 BOA/byte
txsize = 100
if tx fee is 0 - 99 then adjusted fee is 0, so we don't allow the transaction
if tx fee is 100 or above, then adjusted fee is 1+, so we allow the transaction 

minfee = 0 BOA/byte
txsize = 100
if tx fee is 0 - 99 then adjusted fee is 0, so we allow the transaction
if tx fee is 100 or above, then adjusted fee is 1+, so we allow the transaction